inputremapper/injection/consumers/mapping_parser.py

```diff
@@ -0,0 +1,199 @@
+        containing_keys = [
+            og_key for og_key in all_keys if og_key.contains_key(single_key)
+        ]
+        assert len(containing_keys) != 0
+        if len(containing_keys) == 1:
+            # there was only one handler containing that key
+            sorted_handlers[single_key] = handlers[containing_keys[0]]
+            continue
+
+        keys_to_sort = []
+        for og_key in containing_keys:
+            keys_to_sort.append(og_key)
+
+        assert len(keys_to_sort) > 1
+        # rank the keys and create the HierarchyHandler
+        sorted_keys = _order_keys(keys_to_sort, single_key)
+        sub_handlers = []
+        for og_key in sorted_keys:
+            sub_handlers.append(handlers[og_key])
+
+        hierarchy_handler = HierarchyHandler(sub_handlers)
+        sorted_handlers[single_key] = hierarchy_handler
+    return sorted_handlers
+
+
+def _order_keys(keys: List[Key], common_key: Key) -> List[Key]:
+    """reorder the keys according to some rules
+
+    such that a combination a+b+c is in front of a+b which is in front of b
+    for a+b+c vs. b+d+e: a+b+c would be in front of b+d+e, because the common key b
+    has the higher index in the a+b+c (1), than in the b+c+d (0) list
+    in this example b would be the common key
+    as for combinations like a+b+c and e+d+c with the common key c: ¯\_(ツ)_/¯
+
+    Parameters
+    ----------
+    keys : List[Key]
+        the list which needs ordering
+    common_key : Key
+        the Key all members of Keys have in common
+    """
+    keys.sort(key=len, reverse=True)  # sort by descending length
+
+    def idx_of_common_key(_key: Key) -> int:
+        """get the index of the common key in _key"""
+        for j, sub_key in enumerate(_key):
+            logger.debug(f"idx: {j}, sub_key: {sub_key}")
+            if sub_key == common_key:
+                logger.debug(f"return: {j}")
+                return j
+
+    last_key = keys[0]
+    last_idx = 0
+    for i, key in enumerate([*keys[1:], ((None, None, None),)]):
+        i += 1
+        if len(key) == len(last_key):
+            last_key = key
+            continue
+
+        assert len(key) < len(last_key)
+        sub_list = keys[last_idx:i]
+        sub_list.sort(key=idx_of_common_key, reverse=True)
+        keys[last_idx:i] = sub_list
+
+    return keys
+
+
+def _create_new_config(key, symbol_and_target) -> Dict:
+    # TODO: make this obsolete by migrating to new config structure
+    sub_keys = []
+    for sub_key in key:
+        if sub_key[0] != EV_ABS:
+            sub_keys.append(sub_key)
+            continue
+        if sub_key[2] == 1:
+            sub_keys.append((*sub_key[:2], 10))  # trigger point at 10%
+        else:
+            assert sub_key[2] == -1
+            sub_keys.append((*sub_key[:2], -10))  # trigger point at -10%
+    new_key = Key(*sub_keys)
+
+    config = {
+        "key": new_key,
+        "symbol": symbol_and_target[0],
```

Comment:
```suggestion
        "symbol": clean(symbol_and_target[0]),
```

Reply:
on that note: I just inflated the log a bit more.
I created a custom logger which parses a MappingHandler object and spits out the path a event takes through the different handlers. It helps a lot for debugging and understanding the handler interaction:
![image](https://user-images.githubusercontent.com/31804124/150432904-412c19c1-2318-4a3c-961b-1c8ae1b98638.png)
creates this log now:
```
DEBUG mapping_parser.py:75: created mapping handler:
DEBUG mapping_parser.py:76: HierarchyHandler for (1, 310) <140107968623808>:
DEBUG mapping_parser.py:76:     CombinationHandler for ((1, 310, 1), (3, 3, 10)) <140107968622608>:
DEBUG mapping_parser.py:76:         KeyHandler <140107968622800>:
DEBUG mapping_parser.py:76:             maps to: (1, 2) on keyboard
DEBUG mapping_parser.py:76:     CombinationHandler for ((1, 310, 1),) <140107968623424>:
DEBUG mapping_parser.py:76:         KeyHandler <140107968623616>:
DEBUG mapping_parser.py:76:             maps to: (1, 4) on keyboard
DEBUG mapping_parser.py:81: created mapping handler:
DEBUG mapping_parser.py:82: AbsToBtnHandler for (3, 3, 10) <140107968623712>:
DEBUG mapping_parser.py:82:     HierarchyHandler for (3, 3) <140107968624048>:
DEBUG mapping_parser.py:82:         CombinationHandler for ((1, 310, 1), (3, 3, 10)) <140107968622608>:
DEBUG mapping_parser.py:82:             KeyHandler <140107968622800>:
DEBUG mapping_parser.py:82:                 maps to: (1, 2) on keyboard
DEBUG mapping_parser.py:82:         CombinationHandler for ((3, 3, 10),) <140107968622992>:
DEBUG mapping_parser.py:82:             KeyHandler <140107968623184>:
DEBUG mapping_parser.py:82:                 maps to: (1, 3) on keyboard
```